### PR TITLE
resolve  07-async-code-fetch-api

### PR DESCRIPTION
**Pull request by valeriiSkliar**
Hello!
- Column chart, my solution.
- Sortoble table, took your solution. Sorry, but for personal reasons I did not have time to finish my version of the solution.

### CODE REVIEW

**Comment by jsru** [comment]
Добавляю преподавателя (@dosandk) для код-ревью.

File: 07-async-code-fetch-api-part-1/1-column-chart/index.js [hunk 1]
```diff
@@ -3,5 +3,111 @@ import fetchJson from './utils/fetch-json.js';
...
+  } = {}) {
+    this.data = data;
+    this.label = label;
+    this.link = link;
+    this.value = value;
+    this.url = new URL(url, BACKEND_URL);
+    this.range = range;
+    this.formatHeading = formatHeading;
+    this.render();
+    this.update(this.range.from, this.range.to);
```
Comment (dosandk) [inline 07-async-code-fetch-api-part-1/1-column-chart/index.js:29]: 👍 good

File: 07-async-code-fetch-api-part-1/1-column-chart/index.js [hunk 2]
```diff
@@ -3,5 +3,111 @@ import fetchJson from './utils/fetch-json.js';
...
+  }
+
+  async loadData(from, to) {
```
Comment (dosandk) [inline 07-async-code-fetch-api-part-1/1-column-chart/index.js:32]: 👍 good

File: 07-async-code-fetch-api-part-1/1-column-chart/index.js [hunk 3]
```diff
@@ -3,5 +3,111 @@ import fetchJson from './utils/fetch-json.js';
...
+    return this.data.map(item => {
+      const value = String(Math.floor(item * scale));
+      const percent = (item / maxValue * 100).toFixed(0) + '%';
+      const columnProps = `style="--value: ${value}" data-tooltip="${percent}"`;
+      return `<div ${columnProps}></div>`;
+    }).join('');
+  }
+
+  checkDataLength(data = this.data) {
+    if (data.length === 0) {
```
Comment (dosandk) [inline 07-async-code-fetch-api-part-1/1-column-chart/index.js:55]: тут лучше воспользоваться тернарником:
```
return data.length === 0 ? 'column-chart_loading' : ''
```

**Review by dosandk — APPROVED** [review]
good 👍

**Pull request merged**